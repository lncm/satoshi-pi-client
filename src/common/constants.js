import PKG from '../../package.json';

export default {
  NAME: PKG.name,
  VERSION: PKG.version,
};
