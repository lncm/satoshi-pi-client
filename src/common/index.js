import C from './constants';
import STYLE from './style';

export {
  C,
  STYLE,
};
