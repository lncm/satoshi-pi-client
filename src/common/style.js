import { Platform } from 'react-native';

const UNIT = 10;

export default {
  UNIT,
  OFFSET: UNIT * 1.6,

  COLOR: {
    ACCENT: 'black',
    PRIMARY: 'grey',
    WHITE: 'white',
  },

  FONT: {
    DEFAULT: Platform.OS === 'web' ? 'Google Sans,Helvetica,sans-serif;' : undefined,

    TITLE: Platform.OS === 'web' ? 'Product Sans,Helvetica,sans-serif;' : undefined,

    SIZE: {
      REGULAR: UNIT * 1.6,
      TITLE: UNIT * 1.8,
    },

    WEIGHT: {
      REGULAR: '400',
      BOLD: '700',
    },
  },
};
