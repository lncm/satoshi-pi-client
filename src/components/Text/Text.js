import React from 'react';
import { Text as TextNative } from 'react-native';

import styles from './Text.style';

const Text = ({ title, ...inherit }) => (
  <TextNative
    {...inherit}
    style={[styles.regular, title && styles.title, inherit.style]}
  />
);

export default Text;
