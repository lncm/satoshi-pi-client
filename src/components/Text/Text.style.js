import { StyleSheet } from 'react-native';

import { STYLE } from 'common';

const {
  REGULAR, TITLE, SIZE, WEIGHT,
} = STYLE.FONT;

export default StyleSheet.create({
  default: {
    fontFamily: REGULAR,
    fontSize: SIZE.REGULAR,
    fontWeight: WEIGHT.REGULAR,
  },

  title: {
    fontFamily: TITLE,
    fontSize: SIZE.TITLE,
    fontWeight: WEIGHT.BOLD,
  },
});
