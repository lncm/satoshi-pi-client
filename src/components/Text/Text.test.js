import React from 'react';
import renderer from 'react-test-renderer';

import { C } from 'common';
import Text from './Text';

describe('<Text>', () => {
  it('renders', () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {children}', () => {
    const tree = renderer.create(<Text>{C.NAME}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {title}', () => {
    const tree = renderer.create(<Text title>{C.NAME}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

