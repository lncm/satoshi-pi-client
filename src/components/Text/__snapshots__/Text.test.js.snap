// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> renders 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`<Text> when {children} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      undefined,
      undefined,
    ]
  }
>
  satoshi-pi-client
</Text>
`;

exports[`<Text> when {title} 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "fontFamily": undefined,
        "fontSize": 18,
        "fontWeight": "700",
      },
      undefined,
    ]
  }
>
  satoshi-pi-client
</Text>
`;
